sshd-scp/src/main/java/org/apache/sshd/common/scp/ScpHelper.java

```diff
@@ -401,9 +401,11 @@ public void send(Collection<String> paths, boolean recursive, boolean preserve,
                 }
 
                 Session session = getSession();
-                Iterable<String> included = opener.getMatchingFilesToSend(session, basedir, pattern);
+                Path basePath = resolveLocalPath(basedir);
+                Iterable<String> included = opener.getMatchingFilesToSend(session, basePath.toFile().getAbsolutePath(), pattern);
+
                 for (String path : included) {
```

Comment:
Please don't use `toFile` - if you want the absolute path then use `basePath.toAbsolutePath().toString()`.

Reply:
`basePath.toAbsolutePath().toString()` won't give absolute path of the file when the underlying filesystem is virtual filesystem. I couldn't find any other cleaner way other than `toFile`